spec/unit/matrixrtc/MembershipManager.spec.ts

```diff
@@ -76,588 +72,673 @@ const callSession = { id: "", application: "m.call" };
 describe("MembershipManager", () => {
     let client: MockClient;
     let room: Room;
-    const focusActive: LivekitFocusSelection = {
+    const focusActive = Object.freeze({
         focus_selection: "oldest_membership",
         type: "livekit",
-    };
-    const focus: Transport = {
+    });
+    const focus: Transport = Object.freeze({
         type: "livekit",
         livekit_service_url: "https://active.url",
         livekit_alias: "!active:active.url",
-    };
+    });
 
     beforeEach(() => {
         // Default to fake timers.
         jest.useFakeTimers();
         client = makeMockClient("@alice:example.org", "AAAAAAA");
-        room = makeMockRoom([membershipTemplate]);
-        // Provide a default mock that is like the default "non error" server behaviour.
-        (client._unstable_sendDelayedStateEvent as Mock<any>).mockResolvedValue({ delay_id: "id" });
-        (client._unstable_updateDelayedEvent as Mock<any>).mockResolvedValue(undefined);
-        (client._unstable_sendStickyEvent as Mock<any>).mockResolvedValue({ event_id: "id" });
-        (client._unstable_sendStickyDelayedEvent as Mock<any>).mockResolvedValue({ delay_id: "id" });
-        (client.sendStateEvent as Mock<any>).mockResolvedValue({ event_id: "id" });
+        room = makeMockRoom([sessionMembershipTemplate]);
     });
 
     afterEach(() => {
         jest.useRealTimers();
         // There is no need to clean up mocks since we will recreate the client.
     });
-
-    describe("isActivated()", () => {
-        it("defaults to false", () => {
-            const manager = new MembershipManager({}, room, client, callSession);
-            expect(manager.isActivated()).toEqual(false);
+    describe("LegacyMembershipManager", () => {
+        beforeEach(() => {
+            // Provide a default mock that is like the default "non error" server behaviour.
+            (client._unstable_sendDelayedStateEvent as Mock<any>).mockResolvedValue({ delay_id: "id" });
```

Comment:
What is the difference with this notation `jest.mocked(client._unstable_sendDelayedStateEvent).mockResolvedValue({ delay_id: "id" });` ?
That I think I usually see for that